### ISSUE TITLE
Can't infer list type after appending to empty list

### ISSUE THREAD

**Issue by veber-alex** [microsoft/pyright#5766]
```python
def foo(x: list[int]):
    pass


l1 = []
l1.append("sdd")
foo(l1)
```

In "basic" mode this code doesn't report any errors because pyright can't infer the type of the list, even though the act of appending a `str` should set the type of the list to `list[str]`

In "strict" mode this gives an that "Type of "append" is partially unknown".

Workaround is to manually annotate the type of l1 as `l1: list[str] = []`

**Comment by erictraut** [comment]
Yes, this is the intended behavior. The inferred type of `l1` is `list[Unknown]`. Calling `append` on this object doesn't change its type. 

In cases like this, you need to do one of the following:

1. Create a list that's not initially empty.
```python
l1 = ["sdd"]
foo(l1)
```

2. Provide a type declaration to override the default inference rules:

```python
l1: list[str] = []
```

**Comment by veber-alex** [comment]
Understood, thanks for the quick reply.
Though it's unfortunate as mypy handles this correctly.
